src/ui/widgets/EmbeddedDisplay/jsonUtils.ts

```diff
@@ -0,0 +1,97 @@
+import { WidgetDescription } from "../createComponent";
+import { Color } from "../../../types/color";
+import { Font, FontStyle } from "../../../types/font";
+import { GenericProp, Rule } from "../../../types/props";
+import { Border, BorderStyle } from "../../../types/border";
+
+export interface JsonDescription {
+  type: string;
+  x: string;
+  y: string;
+  height: string;
+  width: string;
+  children: JsonDescription[];
+  [key: string]: any;
+}
+function jsonParseColor(name: string, jsonColor: string): Color {
+  return Color.parse(jsonColor);
+}
+
+function jsonParseBorder(name: string, jsonBorder: any): Border {
+  const styles: { [key: string]: BorderStyle } = {
+    none: BorderStyle.None,
+    line: BorderStyle.Line,
+    dashed: BorderStyle.Dashed,
+    dotted: BorderStyle.Dotted,
+    groupbox: BorderStyle.GroupBox
+  };
+  return new Border(
+    styles[jsonBorder.style?.toLowerCase()],
```

Comment:
This is optionally chained, but style is required for the Border type. What style will you get if none is provided? Should this raise an error?

Reply:
Good question, this was probably just a shortcut I took.